Source/Core/VideoBackends/D3D/VertexManager.cpp

```diff
@@ -182,6 +182,13 @@ void VertexManager::vFlush(bool useDstAlpha)
 
 	PrepareDrawBuffers(stride);
 
+	if (!bpmem.genMode.zfreeze)
```

Comment:
I think it should be even fine to move this in the caller class within VideoCommon. But you have to search for the stride yourself at this point. However, PrepareDrawBuffers isn't needed if we want to drop the draw call.